src/omnicore/mdex.cpp

```diff
@@ -679,7 +666,7 @@ bool mastercore::MetaDEx_isOpen(const uint256& txid, uint32_t propertyIdForSale)
 // to save doing a second levelDB iteration if already done in calling function, pass in optional totalSold & totalBought
 std::string mastercore::MetaDEx_getStatus(const uint256& txid, uint32_t propertyIdForSale, int64_t amountForSale, int64_t totalSold, int64_t totalReceived)
 {
-    if (totalSold == -1 || totalReceived == -1) {
+    if (totalSold == -1 || totalReceived == -1) { // TODO: do we need this?
```

Comment:
Yeah, it's a bypass - basically we need the values for ```totalSold``` & ```totalReceived``` but they may already be available to us by being passed in.  The default values are -1 so if they're still at defaults we know we haven't been passed the necessary info and must go and fetch it via ```getMatchingTrades```, if they're not -1 though we know we've already been passed the info and can skip this step.

Hope that makes sense :)